lib/cocoapods-core/specification/linter.rb

```diff
@@ -394,6 +394,22 @@ def _validate_test_type(t)
         end
       end
 
+      def _validate_app_host_name(n)
+        unless consumer.requires_app_host?
+          results.add_error('app_host_name', '`requires_app_host` must be set to ' \
+            '`true` when `app_host_name` is specified.')
+        end
+
+        parent_spec_name = (consumer.spec.parent && consumer.spec.parent.name) || consumer.name
+        podspec_name, appspec_name = n.split('/')
+
+        unless podspec_name == parent_spec_name || consumer.dependencies.map(&:name).include?(appspec_name)
```

Comment:
This doesn't work when depending upon an app spec in another pod, since the dependency will contain the fully-qualified name rather than just the name of the appspec